pkg/cloud/vsphere/provisioner/govmomi/create.go

```diff
@@ -485,19 +484,23 @@ func (pv *Provisioner) getCloudProviderConfig(cluster *clusterv1.Cluster, machin
 	if err != nil {
 		return "", err
 	}
+
+	cpc := vpshereprovisionercommon.CloudProviderConfigTemplate{
+		Datacenter:   machineconfig.MachineSpec.Datacenter,
+		Server:       clusterConfig.VsphereServer,
+		Insecure:     true, // TODO(ssurana): Needs to be a user input
+		UserName:     clusterConfig.VsphereUser,
+		Password:     clusterConfig.VspherePassword,
+		ResourcePool: machineconfig.MachineSpec.ResourcePool,
+		Datastore:    machineconfig.MachineSpec.Datastore,
+		Network:      "",
+	}
+	if len(machineconfig.MachineSpec.Networks) > 0 {
+		cpc.Network = machineconfig.MachineSpec.Networks[0].NetworkName
+	}
+
 	// TODO(ssurana): revisit once we solve https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/issues/60
```

Comment:
move this comment to line#487 since this was essentially meant for the code block above